component/src/main/java/org/wso2/extension/siddhi/io/report/sink/ReportSink.java

```diff
@@ -0,0 +1,570 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.extension.siddhi.io.report.sink;
+
+import org.apache.log4j.Logger;
+import org.wso2.extension.siddhi.io.report.generators.DynamicReportGenerator;
+import org.wso2.extension.siddhi.io.report.generators.QueryModeReportGenerator;
+import org.wso2.extension.siddhi.io.report.generators.StaticReportGenerator;
+import org.wso2.extension.siddhi.io.report.util.ReportConstants;
+import org.wso2.siddhi.annotation.Example;
+import org.wso2.siddhi.annotation.Extension;
+import org.wso2.siddhi.annotation.Parameter;
+import org.wso2.siddhi.annotation.util.DataType;
+import org.wso2.siddhi.core.config.SiddhiAppContext;
+import org.wso2.siddhi.core.event.Event;
+import org.wso2.siddhi.core.exception.ConnectionUnavailableException;
+import org.wso2.siddhi.core.exception.SiddhiAppCreationException;
+import org.wso2.siddhi.core.stream.output.sink.Sink;
+import org.wso2.siddhi.core.util.config.ConfigReader;
+import org.wso2.siddhi.core.util.transport.DynamicOptions;
+import org.wso2.siddhi.core.util.transport.OptionHolder;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.definition.StreamDefinition;
+
+import java.io.File;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This class contains the implementation of siddhi-io-file sink which provides the functionality of publishing data
+ * to reports as pdf files through siddhi.
+ */
+
+@Extension(
+        name = "report",
+        namespace = "sink",
+        description = "Report sink can be used to publish (write) event data which is processed within siddhi" +
```

Comment:
Shall we add more information regarding the extension? Because this will be used to generate the docs which is used by the external users

Reply:
Fixed with 1b03449baa893cb7e22553cab0a8bf1298544c42